File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  let newStr = ''
  for (let i = str.length - 1; i >= 0; i--) {
    newStr += str[i]
  }
  return newStr
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("Expecting: 'eissej'");
  console.log("=>", reverseString('jessie'))

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
  // initialize an empty string called 'newStr'
  // iterate over each character in the string from back to front
    // push the last character in the string onto the newStr
  // return the newStr

// And a written explanation of your solution
  // The function accepts a string as an argument. A variable called newStr is initialized and set to an empty string. 
  // A for loop is used to iterate over the array, starting at the end, until there are no more characters in the array. 
  // The characters are added to the empty newStr in reverse order using the push method. Then newStr is returned.

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  reversed = []
  string = str.split('')
  string.each do |char|
    reversed.unshift(char)
  end
  return reversed.join('')

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts "Expecting: 'eissej'"
  puts "=>", reverse_string('jessie')

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
  # initialize an empty array called reversed
  # set a variable called string equal to the str argument split into an array
  # iterate over the string array and push each character to the beginning of the reversed array
  # return the reversed array joined as a string
# And a written explanation of your solution
  # A reversed variable is declared and set to an empty array. The string argument is split into an array and stored in the 'string' variable. The 'string' array is iterated over, and each item is pushed to the front of the reversed array using unshift, which will result in a reversed array. The 'reversed' array is joined as a string and returned.